ports/abseil/portfile.cmake

```
@@ -44,16 +41,20 @@ file(REMOVE_RECURSE "${CURRENT_PACKAGES_DIR}/debug/share"
                     "${CURRENT_PACKAGES_DIR}/include/absl/time/internal/cctz/testdata"
 )
 
-if (VCPKG_LIBRARY_LINKAGE STREQUAL dynamic)
-    vcpkg_replace_string("${CURRENT_PACKAGES_DIR}/include/absl/base/config.h"
-        "#elif defined(ABSL_CONSUME_DLL)" "#elif 1"
-    )
-    vcpkg_replace_string("${CURRENT_PACKAGES_DIR}/include/absl/base/internal/thread_identity.h"
-        "&& !defined(ABSL_CONSUME_DLL)" "&& 0"
-    )
-    vcpkg_replace_string("${CURRENT_PACKAGES_DIR}/include/absl/container/internal/hashtablez_sampler.h"
+function(fix_dll_definitions file)
+    vcpkg_replace_string("${file}"
         "!defined(ABSL_CONSUME_DLL)" "0"
     )
+    vcpkg_replace_string("${file}"
+        "defined(ABSL_CONSUME_DLL)" "1"
+    )
+endfunction(fix_dll_definitions file)
+
+if (VCPKG_LIBRARY_LINKAGE STREQUAL dynamic)
+    file(GLOB_RECURSE headers "${CURRENT_PACKAGES_DIR}/include/absl/*.h")
+    foreach(header ${headers})
+        fix_dll_definitions("${header}")

```

Can't this be a straight
```suggestion
        vcpkg_replace_string("${header}" "defined(ABSL_CONSUME_DLL)" "(1)")
```